src/ucp/rndv/rndv_put.c

```diff
@@ -126,26 +126,34 @@ ucp_proto_rndv_put_common_atp_send(ucp_request_t *req, ucp_lane_index_t lane)
 {
     const ucp_proto_rndv_put_priv_t *rpriv = req->send.proto_config->priv;
     ucp_proto_rndv_put_atp_pack_ctx_t pack_ctx;
+    ucs_status_t status;
 
     pack_ctx.req = req;
 
+    if (req->send.rndv.put.atp_count >= req->send.state.dt_iter.length) {
```

Comment:
1. Do I understand correctly that there is no correctness problem in calculating `atp_sent_map`(and therefore `atp_sent_count`) as `rpriv->atp_map & UCS_MASK(request->send.rndv.put.atp_lane);` and you introduced `atp_count` request field mostly for performance purposes?
2. I understand that this if does its job, but AFAIK it works only for `min_length==1`. So I would suggest to create a more general case when `atp_count` compared with `dt_iter.length / min_length`. Or maybe add assertion that `min_length == 1` and probably add some comment to explain what does this if do.

Reply:
1. atp_count was added to check the actual number of atp messages sent, so it will not exceed "length". if we counted it using "lane" and mask, it would involve popcount bit operation which is expensive
2. unfortunately, setting min_length to 1 exposes issue with incorrect handling of overlap on sender side (when receiving RTR). so have to keep min_length==0 for rndv/put protocol. so IMO we can keep the same behavior as the previous code - which is to ignore 0-length ATP - and set min_length=1 when the overlap issue is fixed